src/main/java/gregtech/api/items/toolitem/IGTTool.java

```diff
@@ -552,6 +553,22 @@ default AoESymmetrical getAoEDefinition(ItemStack stack) {
         }
     }
 
+    default boolean definition$canApplyAtEnchantingTable(ItemStack stack, Enchantment enchantment) {
+        if(stack.isEmpty()) {
+            return false;
+        }
+
+        ToolProperty property = getToolProperty(stack);
+
+        if(property == null)  {
+            return false;
+        }
+
+        // Check for any special enchantments specified by the material of this Tool, or any additional Enchantment Types added in the builder
+        return (!property.getEnchantments().isEmpty() && property.getEnchantments().containsKey(enchantment)) ||
```

Comment:
Do we need to check if the property's enchantments are empty? The `containsKey` method I believe returns false for an empty map.